lua/entities/gmod_wire_expression2/core/custom/sv_rune2.lua

```diff
@@ -20,69 +21,54 @@ local function buildBody(args) -- Why does the wireteam do this
 end
 
 -- We use this for try and catch
-local function runE2InstanceSafe(compiler,func,body) -- Varargs to pass to the e2 function
-    -- Will always return pcallerror,errstr first even if it didn't error.
-    local args = {pcall(func,compiler,body)}
+local function runE2InstanceSafe(compiler,func,body,...)
+    -- Always return pcallerror,errstr first even if it didn't error.
+    local args = {pcall(func,compiler,body,...)}
     local success = table_remove(args,1)
     if success then
-        return true,"none",args
-    else
-        -- Don't return args, that would be a waste
-        local errmsg = table_remove(args,1)
-        if errmsg == "exit" then return true,"none",{} end -- nice exit()
-        if errmsg == "perf" then errmsg = "tick quota exceeded" end
-        return false,errmsg
+        return true,nil,args
     end
+    -- If unsuccessful, don't return args, that would be a waste
+    local errmsg = table_remove(args,1)
+    if errmsg == "exit" then return true,nil,{} end -- nice exit()
+    if errmsg == "perf" then errmsg = "tick quota exceeded" end
+    return false,errmsg
 end
 
-local function getE2FuncFromStr(compiler,funcname)
```

Comment:
Same as earlier, this function has been refactored and moved into the single place, into helpers file.
Also, I forgot to say, I have also fixed that pattern as well.
It was failing to find E2 UDF if the name contained underscore(s), for example `h_3llo_` is a valid UDF name.